docs(flutter): Document `beforeCapture` function customization (#12190)
getsentry/sentry-docs: docs/platforms/flutter/enriching-events/viewhierarchy/index.mdx

<PlatformContent includePath="enriching-events/attach-viewhierarchy" />

### Customize View Hierarchy Capturing

<Note>

> Requires SDK version `8.12.0` or higher.

Reviewer: we released `8.12.0` already

```suggestion
Requires SDK version `8.13.0` or higher.
```

</Note>

> Because capturing view hierarchy can be a resource-intensive operation on Flutter, it's limited to one view hierarchy every 2 seconds using a debouncing mechanism. This behavior can be overruled if you supply a `BeforeCaptureCallback` for view hierarchies in the `SentryFlutterOptions`.

Reviewer: ```suggestion
Capturing view hierarchies on Flutter is limited to once every 2 seconds by default to minimize performance impact. While this debounce interval is fixed, you can override individual capture decisions by implementing the `beforeCaptureViewHierarchy` option in your `SentryFlutterOptions`.
```

> The `BeforeCaptureCallback` also allows you to customize behavior based on event data so you can decide when to capture view hierarchy and when not to. For example, you can decide to only capture view hierarchy for fatal events:

Reviewer: ```suggestion
The `beforeCaptureViewHierarchy` option allows you to customize behavior based on event data so you can decide when to capture view hierarchy and when not to. For example, you can decide to only capture view hierarchy for fatal events:
```

> ```java

Reviewer: ```suggestion
```dart
```

pls add line highlighting

await SentryFlutter.init((options) {

>   options.attachViewHierarchy = true;

Reviewer: ```suggestion
```

  options.beforeCaptureViewHierarchy = (event, hint, debounce) async {
    // If debounce is active, skip capturing
    if (debounce) {
      return false;

>     }

Reviewer: iirc that means that users can also skip debouncing if they return true anyway, right?

Author: Correct, it's the same API that we have on Android. it's also mentioned in the docs AFAIK.
